rotkehlchen/chain/ethereum/modules/weth/decoder.py

```diff
@@ -0,0 +1,153 @@
+                event.counterparty = CPT_WETH
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Deposit {deposited_amount} {self.eth.symbol} for WETH'  # noqa: E501
+
+        event = HistoryBaseEntry(
+            event_identifier=transaction.tx_hash,
+            sequence_index=self.base_tools.get_next_sequence_counter(),
+            timestamp=ts_sec_to_ms(transaction.timestamp),
+            location=Location.BLOCKCHAIN,
+            event_type=HistoryEventType.RECEIVE,
+            event_subtype=HistoryEventSubType.RECEIVE_WRAPPED,
+            asset=A_WETH,
+            balance=Balance(amount=deposited_amount),
+            location_label=depositor,
+            counterparty=CPT_WETH,
+            notes=f'Receive {deposited_amount} {self.weth.symbol}',
+        )
+        return event, None
+
+    def _decode_withdrawal_event(
+            self,
+            tx_log: EthereumTxReceiptLog,
+            transaction: EvmTransaction,
+            decoded_events: List[HistoryBaseEntry],
+            all_logs: List[EthereumTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: Optional[List[ActionItem]],  # pylint: disable=unused-argument
+    ) -> Tuple[Optional[HistoryBaseEntry], Optional[ActionItem]]:
+        withdrawer = hex_or_bytes_to_address(tx_log.topics[1])
+        withdrawed_amount_raw = hex_or_bytes_to_int(tx_log.data[:32])
+        withdrawed_amount = asset_normalized_value(amount=withdrawed_amount_raw, asset=self.eth)
+
+        in_event = None
+        for event in decoded_events:
+            if (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.counterparty in (WETH_CONTRACT, withdrawer) and
+                event.balance.amount == withdrawed_amount and
+                event.asset == self.eth
+            ):
+                in_event = event
+                event.counterparty = CPT_WETH
+                event.notes = f'Receive {withdrawed_amount} ETH'
+
+        weth_withdrawal_event = HistoryBaseEntry(
+            event_identifier=transaction.tx_hash,
+            timestamp=ts_sec_to_ms(transaction.timestamp),
+            sequence_index=self.base_tools.get_next_sequence_counter(),
+            location=Location.BLOCKCHAIN,
+            event_type=HistoryEventType.WITHDRAWAL,
+            event_subtype=HistoryEventSubType.REMOVE_ASSET,
+            asset=A_WETH,
+            balance=Balance(amount=withdrawed_amount),
+            location_label=withdrawer,
+            counterparty=CPT_WETH,
+            notes=f'Withdraw {withdrawed_amount} WETH for ETH',
+        )
+        maybe_reshuffle_events(
+            out_event=weth_withdrawal_event,
+            in_event=in_event,
+            events_list=decoded_events + [weth_withdrawal_event],
+        )
```

Comment:
Is this reorg needed? Does it matter the order here?

Reply:
I suppose. Given that unwrapping happens before the `ETH` is sent